### PR TITLE
Expose low-level diagnostics functionality

### PR DESCRIPTION
**Pull request by pyhys**
This is requested by users that use p-net for testing of
IO-controller applications.

Add configuration flag for default diagnosis mechanism
Also improve argument validation to diagnosis functions.

### CODE REVIEW

File: test/utils_for_testing.cpp [hunk 1]
```diff
@@ -547,6 +547,9 @@ void PnetIntegrationTestBase::cfg_init()
 
    /* Storage */
    strcpy (pnet_default_cfg.file_directory, "/disk1");
+
+   /* Diagnosis */
+   pnet_default_cfg.use_qualified_diagnosis = true;
```
Comment (olbjo) [inline test/utils_for_testing.cpp:552]: I can't see that this config is set for the sample application. I think it should.

**Review by olbjo — APPROVED** [review]
I think the commit message could describe why the api was changed. It is not clear to me.

File: test/utils_for_testing.cpp (hunk 1 above)
Comment (pyhys) [reply test/utils_for_testing.cpp:552]: Done

**Comment by pyhys** [comment]
Improved the commit message.

File: src/device/pf_diag.c [hunk 2]
```diff
@@ -720,22 +819,22 @@ int pf_diag_remove (
                    * the specifier PNET_DIAG_CH_SPEC_ALL_DISAPPEARS is used. */
                   /* Overwrite the item to remove with correct cr_properties for
                    * this diag message. */
+                  /* Why change USI? */
```
Comment (hefloryd) [inline src/device/pf_diag.c:822]: What does this comment refer to?

**Review by hefloryd — APPROVED** [review]

File: src/device/pf_diag.c (hunk 2 above)
Comment (pyhys) [reply src/device/pf_diag.c]: It was a note to myself, now properly removed.

**Pull request merged**